File: Java Barcode Scanner ME/app/src/main/java/de/dimedis/mobileentry/backend/response/PerformEntryCheckoutResponseContent.java
package de.dimedis.mobileentry.backend.response;

import androidx.annotation.NonNull;

import com.google.gson.annotations.SerializedName;

import java.util.List;

import de.dimedis.lmlib.myinterfaces.Access;
import de.dimedis.mobileentry.model.Action;
import de.dimedis.mobileentry.model.Ticket;

public class PerformEntryCheckoutResponseContent extends BaseResponseContent {
    @SerializedName("status_modifiers")
    public List<String> statusModifiers;
    @SerializedName("ticket_code")
    public String ticketCode;
    @SerializedName("deny_reason")
    public String denyReason;
    public Ticket ticket;
    public Action action;

    public static PerformEntryCheckoutResponseContent fromLmLib(@NonNull Access access) {
        PerformEntryCheckoutResponseContent content = new PerformEntryCheckoutResponseContent();
        content.status = access.getStatus();
        content.statusDescription = access.getStatusDescription();
        content.userMessage = access.getUserMessage();
        content.ticketCode = access.getTicketCode();
        content.denyReason = access.getDenyReason();
        content.ticket = Ticket.fromLmLib(access.getTicket());
        content.action = Action.fromLmLib(access.getAction());
        content.statusModifiers = access.getStatusModifiers();
        return content;
    }

    @Override
    public String toString() {
        return "PerformEntryCheckoutResponseContent{" +
                "statusModifiers=" + statusModifiers +
                ", ticketCode='" + ticketCode + '\'' +
                ", denyReason='" + denyReason + '\'' +
                ", ticket=" + ticket +
                ", action=" + action +
                '}';
    }

    public boolean isReducedTicket() {
        return statusModifiers != null && statusModifiers.contains("reduced");
    }
}

File: Kotlin Migration/NTMobileEntry/app/src/main/java/de/dimedis/mobileentry/backend/response/PerformEntryCheckoutResponseContent.kt
package de.dimedis.mobileentry.backend.response

import com.google.gson.annotations.SerializedName
import de.dimedis.mobileentry.model.Action
import de.dimedis.mobileentry.model.Ticket

object PerformEntryCheckoutResponseContent : BaseResponseContent() {
    @SerializedName("status_modifiers")
    var statusModifiers: List<String>? = null

    @SerializedName("ticket_code")
    var ticketCode: String? = null

    @SerializedName("deny_reason")
    var denyReason: String? = null
    var ticket: Ticket? = null
    var action: Action? = null

    fun fromLmLib(): PerformEntryCheckoutResponseContent {
        val content = PerformEntryCheckoutResponseContent
        content.status = status
        content.statusDescription = statusDescription
        content.userMessage = userMessage
        content.ticketCode = ticketCode
        content.denyReason = denyReason
        content.ticket = Ticket.fromLmLib(ticket!!)
        content.action = Action.fromLmLib(action)
        content.statusModifiers = statusModifiers
        return content
    }

    override fun toString(): String {
        return "PerformEntryCheckoutResponseContent{" +
                "statusModifiers=" + statusModifiers +
                ", ticketCode='" + ticketCode + '\'' +
                ", denyReason='" + denyReason + '\'' +
                ", ticket=" + ticket +
                ", action=" + action +
                '}'
    }

    fun isReducedTicket(): Boolean {
        return statusModifiers != null && statusModifiers!!.contains("reduced")
    }

}